sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AADOAuth2OboAuthorizedClientRepository.java

```diff
@@ -0,0 +1,148 @@
+        this.azureClientRegistrationRepository = azureClientRegistrationRepository;
+        Iterator<ClientRegistration> iterator = azureClientRegistrationRepository.iterator();
+        while (iterator.hasNext()) {
+            ClientRegistration next = iterator.next();
+            this.confidentialClientApplicationMap.put(next.getRegistrationId(), createApp(next));
+        }
+    }
+
+    @Override
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public <T extends OAuth2AuthorizedClient> T loadAuthorizedClient(String registrationId,
+                                                                     Authentication authentication,
+                                                                     HttpServletRequest request) {
+        try {
+            String oboAuthorizedClientAttributeName = OBO_AUTHORIZEDCLIENT_PREFIX + registrationId;
+            if (request.getAttribute(oboAuthorizedClientAttributeName) != null) {
+                return (T) request.getAttribute(oboAuthorizedClientAttributeName);
+            }
+
+            if (!(authentication instanceof AbstractOAuth2TokenAuthenticationToken)) {
+                throw new IllegalStateException("Not support token implementation");
+            }
+            AbstractOAuth2TokenAuthenticationToken<AbstractOAuth2Token> authenticationToken =
+                (AbstractOAuth2TokenAuthenticationToken) authentication;
+            ClientRegistration clientRegistration =
+                azureClientRegistrationRepository.findByRegistrationId(registrationId);
+
+            String accessToken = authenticationToken.getToken().getTokenValue();
+            OnBehalfOfParameters parameters = OnBehalfOfParameters
+                .builder(clientRegistration.getScopes(), new UserAssertion(accessToken))
+                .build();
+            ConfidentialClientApplication clientApplication =
+                getClientApplication(clientRegistration.getRegistrationId());
+            String oboAccessToken = clientApplication.acquireToken(parameters).get().accessToken();
+
+            JWT parser = JWTParser.parse(oboAccessToken);
+            Date iat = (Date) parser.getJWTClaimsSet().getClaim("iat");
+            Date exp = (Date) parser.getJWTClaimsSet().getClaim("exp");
+            OAuth2AccessToken oAuth2AccessToken = new OAuth2AccessToken(OAuth2AccessToken.TokenType.BEARER,
+                oboAccessToken,
+                Instant.ofEpochMilli(iat.getTime()),
+                Instant.ofEpochMilli(exp.getTime()));
+
+            OAuth2AuthorizedClient oAuth2AuthorizedClient = new OAuth2AuthorizedClient(clientRegistration,
+                authenticationToken.getName(), oAuth2AccessToken);
+
+            request.setAttribute(oboAuthorizedClientAttributeName, (T) oAuth2AuthorizedClient);
+            return (T) oAuth2AuthorizedClient;
+        } catch (Throwable throwable) {
+            LOGGER.error("Failed to loadAuthorizedClient", throwable);
```

Comment:
change `loadAuthorizedClient` to `load authorized client`, better log the registrationId too in the error message